src/main/index.ts

```diff
@@ -77,6 +76,12 @@ async function main() {
   windowManager = new WindowManager(proxyPort);
 }
 
+// Disable hardware acceleration if flag is checked in Preferences
+userStore.load();
+if (userStore.preferences.disableHardwareAcceleration) {
```

Comment:
This should be handled automatically as a mobx reaction inside user-store.

Reply:
The thing is `app.disableHardwareAcceleration();` should be called before app's `ready` event (https://www.electronjs.org/docs/all#appdisablehardwareacceleration)